rotkehlchen/tests/unit/decoders/test_weth.py

```diff
@@ -465,3 +490,581 @@ def test_wxdai_wrap(database, gnosis_inquirer, gnosis_accounts):
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+            asset=A_WETH_SCROLL,
+            balance=Balance(amount=FVal(0.135)),
+            location_label='0xdFd21F8aA81c5787160F9a4B39357F5FE1c743DC',
+            notes=f'Receive {amount} WETH',
+            counterparty=CPT_WETH,
+            address=string_to_evm_address('0x5300000000000000000000000000000000000004'),
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('base_accounts', [['0x44f29ebE386c409376C66ad268F9Ae595c8C3e76']])
+def test_weth_withdraw_base(database, base_inquirer):
+    evmhash = deserialize_evm_tx_hash('0x8d54608c2f684d880ad40a16cf9b82525c51520798ae8875d543d3338327ddad')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=base_inquirer,
+        database=database,
+        tx_hash=evmhash,
+    )
+    timestamp = TimestampMS(1712239837000)
+    amount, gas_fees = '0.00022448658511341', '0.000000533995613184'
+    expected_events = [
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.BASE,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(gas_fees)),
+            location_label='0x44f29ebE386c409376C66ad268F9Ae595c8C3e76',
+            notes=f'Burned {gas_fees} ETH for gas',
+            counterparty=CPT_GAS,
+        ),
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.BASE,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.RETURN_WRAPPED,
+            asset=A_WETH_BASE,
+            balance=Balance(amount=FVal(amount)),
+            location_label='0x44f29ebE386c409376C66ad268F9Ae595c8C3e76',
+            notes=f'Unwrap {amount} WETH',
+            counterparty=CPT_WETH,
+            address=string_to_evm_address('0x4200000000000000000000000000000000000006'),
+        ),
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=2,
+            timestamp=timestamp,
+            location=Location.BASE,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(amount)),
+            location_label='0x44f29ebE386c409376C66ad268F9Ae595c8C3e76',
+            notes=f'Receive {amount} ETH',
+            counterparty=CPT_WETH,
+            address=string_to_evm_address('0x4200000000000000000000000000000000000006'),
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('base_accounts', [['0xf396e7dbb20489D47F2daBfDA013163223B892a0']])
+def test_weth_deposit_base(database, base_inquirer):
```

Comment:
you already did them but there is no need to tests **every** case for every chain if they are just repeating the logic across other evm chains.

Just 1 test per chain if other cases are repeating.